BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,395 @@
+
+    f.write(reinterpret_cast<const char*>(image_data.data()),
+            sizeof(float) * image_data.size());
+}
+
+void Render::SetLight(const std::vector<LightInfo>& light_settings)
+{
+    for (const auto& light: light_settings)
+    {
+        Light::Ptr light_instance;
+
+        if (light.type == "point")
+        {
+            light_instance = PointLight::Create();
+        }
+        else if (light.type == "direct")
+        {
+            light_instance = DirectionalLight::Create();
+        }
+        else if (light.type == "spot")
+        {
+            light_instance = SpotLight::Create();
+            SpotLight::Ptr spot = std::dynamic_pointer_cast<SpotLight>(light_instance);
+            spot->SetConeShape(light.cs);
+        }
+        else if (light.type == "ibl")
+        {
+            light_instance = ImageBasedLight::Create();
+
+            ImageBasedLight::Ptr ibl = std::dynamic_pointer_cast<
+                ImageBasedLight>(light_instance);
+
+            auto image_io(ImageIo::CreateImageIo());
+            // check that texture file is exist
+            auto texure_path = std::filesystem::path(light.texture);
+
+            if (!std::filesystem::exists(texure_path))
+            {
+                THROW_EX("textrue image doesn't exist on specified path")
+            }
+
+            Texture::Ptr tex = image_io->LoadImage(light.texture);
+            ibl->SetTexture(tex);
+            ibl->SetMultiplier(light.mul);
+        }
+        else
+        {
+            THROW_EX("unsupported light type")
+        }
+
+        light_instance->SetPosition(light.pos);
+        light_instance->SetDirection(light.dir);
+        light_instance->SetEmittedRadiance(light.rad);
+        m_scene->AttachLight(light_instance);
+    }
+}
+
+void Render::GenerateDataset(const std::vector<CameraInfo>& camera_states,
+                             const std::vector<LightInfo>& light_settings,
+                             const std::vector<int>& spp,
+                             const std::filesystem::path& output_dir,
+                             bool gamma_corection_enabled)
+{
+    using namespace RadeonRays;
+
+    if (!std::filesystem::is_directory(output_dir))
+    {
+        THROW_EX("incorrect output directory signature");
+    }
+
+    // check if number of samples to render wasn't specified
+    if (spp.empty())
+    {
+        return;
+    }
+
+    const int iter_number = *(std::max_element(spp.begin(), spp.end()));
+
+    SetLight(light_settings);
+
+    int cam_index = 1;
+    for (const auto &cam_state: camera_states)
+    {
+        // create camera if it wasn't  done earlier
+        if (!m_camera)
+        {
+            m_camera = Baikal::PerspectiveCamera::Create(cam_state.at,
+                                                         cam_state.pos,
+                                                         cam_state.up);
+
+            m_camera->SetSensorSize(RadeonRays::float2(0.036f, 0.036f));
```

Comment:
Sensor size aspect ratio should be recalculated proportional to target image aspect ratio

Reply:
Done